src/shortcuts/redo.ts

```diff
@@ -1,20 +1,36 @@
-import { Dispatch } from 'react'
 import RedoIcon from '../components/RedoIcon'
-import { Shortcut } from '../@types'
+import { Patch, Shortcut } from '../@types'
 import { isRedoEnabled } from '../selectors/isRedoEnabled'
+import { alert as alertAction } from '../action-creators'
+import { NAVIGATION_ACTIONS } from '../constants'
+import { startCase } from 'lodash'
 
-interface RedoAction {
-  type: 'redoAction'
+/**
+ * Recursively calculates last action type from patches history if it is one of the navigation actions and finally returns the action.
+ */
+const getLatestActionType = (patches: Patch[], n = 1): string => {
+  const lastActionType = patches[patches.length - n]?.[0]?.actions[0]
+  if (NAVIGATION_ACTIONS[lastActionType]) return getLatestActionType(patches, n + 1)
+  return lastActionType
```

Comment:
It's a little cleaner to do a single return statement here, but no worries.

```ts
return NAVIGATION_ACTIONS[lastActionType] ? getLatestActionType(patches, n + 1) : lastActionType
```

Reply:
Sure will keep in mind from next time!